Purchases/Public/RCPurchases.m

```diff
@@ -599,49 +569,51 @@ - (void)updateOriginalApplicationVersionWithCompletionBlock:(RCReceivePurchaserI
                             introductoryPrice:nil
                                  currencyCode:nil
                                    completion:^(RCPurchaserInfo * _Nullable info, NSError * _Nullable error) {
-                                       [self handleUpdatedPurchaserInfo:info error:error];
+                                       [self cachePurchaserInfo:info];
                                        if (completion) {
-                                           completion(info, error);
+                                           [self dispatch:^{
+                                               CALL_AND_DISPATCH_IF_SET(completion, info, error);
+                                           }];
                                        }
                                    }];
             }];
         }
     }];
 }
 
+- (void)clearCaches {
+    [self.userDefaults removeObjectForKey:RCAppUserDefaultsKey];
```

Comment:
I think this should be `removeObjectForKey:self.purchaserInfoUserDefaultCacheKey`. `RCAppUserDefaultsKey` is used to cache the AppUserID.